From 40074f9217b91bcca1102af8c623134cb145db62 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Fri, 22 Sep 2017 22:19:02 +0800
Subject: [PATCH 2583/5242] MLK-16536-9 video: fbdev: dcss: remove undesired
 variables in 'dcss_set_par()'

commit  e858a307b78628dce573ab99fc5f4d89de1ef16d from
https://source.codeaurora.org/external/imx/linux-imx.git

The variables 'cinfo' and 'chan_info' both refer to the
same 'struct dcss_channel_info' data. So remove 'chan_info'
and its related variables to make code more clean.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Reviewed-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |   11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index a616c2d..5b54c14 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -2613,21 +2613,16 @@ static int dcss_set_par(struct fb_info *fbi)
 	int fb_node = fbi->node;
 	struct dcss_channel_info *cinfo = fbi->par;
 	struct dcss_info *info = cinfo->dev_data;
-	struct dcss_channels *chans = &info->chans;
-	struct dcss_channel_info *chan_info;
-	struct cbuffer *cb;
+	struct cbuffer *cb = &cinfo->cb;
 	struct ctxld_commit *cc;
 
 	if (fb_node < 0 || fb_node > 2)
 		BUG_ON(1);
 
-	chan_info = &chans->chan_info[fb_node];
-	cb = &chan_info->cb;
-
 	/* TODO: add save/recovery when config failed */
-	fb_var_to_pixmap(&chan_info->input, &fbi->var);
+	fb_var_to_pixmap(&cinfo->input, &fbi->var);
 
-	ret = config_channel_pipe(chan_info);
+	ret = config_channel_pipe(cinfo);
 	if (ret)
 		goto fail;
 
-- 
1.7.9.5

